stellargraph/data/explorer.py

```diff
@@ -546,54 +546,49 @@ def run(self, nodes, *, n=None, length=None, metapaths=None, seed=None):
-            label = self.graph.node_type(node, use_ilocs=True)
-            filtered_metapaths = [
-                metapath
-                for metapath in metapaths
-                if len(metapath) > 0 and metapath[0] == label
-            ]
+            node_type = self.graph.node_type(node, use_ilocs=True)
+            filtered_metapaths = metapaths_for_node_type[node_type]
 
             for metapath in filtered_metapaths:
-                # augment metapath to be length long
-                # if (
-                #     len(metapath) == 1
-                # ):  # special case for random walks like in a homogeneous graphs
-                #     metapath = metapath * length
-                # else:
-                metapath = metapath[1:] * ((length // (len(metapath) - 1)) + 1)
                 for _ in range(n):
-                    walk = (
-                        []
-                    )  # holds the walk data for this walk; first node is the starting node
+                    # holds the walk data for this walk; first node is the starting node
                     current_node = node
-                    for d in range(length):
-                        walk.append(current_node)
-                        # d+1 can also be used to index metapath to retrieve the node type for the next step in the walk
-                        neighbours = self.graph.neighbors(current_node, use_ilocs=True)
-                        # filter these by node type
-                        neighbours = [
-                            n_node
-                            for n_node in neighbours
-                            if self.graph.node_type(n_node, use_ilocs=True)
-                            == metapath[d]
-                        ]
+                    walk = [current_node]
+                    for target_node_type in metapath:
+                        neighbours = self.graph.neighbors(
+                            current_node,
+                            other_node_type=target_node_type,
+                            use_ilocs=True,
+                        )
+
+                        # if no neighbours of the required type as dictated by the metapath exist, then stop.
                         if len(neighbours) == 0:
-                            # if no neighbours of the required type as dictated by the metapath exist, then stop.
                             break
+
                         # select one of the neighbours uniformly at random
-                        current_node = rs.choice(
-                            neighbours
-                        )  # the next node in the walk
+                        current_node = rs.choice(neighbours)
+                        walk.append(current_node)
 
                     walks.append(
                         list(self.graph.node_ilocs_to_ids(walk))
                     )  # store the walk
 
         return walks
 
+    def _augment_and_group_metapaths(self, metapaths, walk_length):
```

Comment:
What does "augment" mean here? Maybe:

```suggestion
    def _extend_and_group_metapaths(self, metapaths, walk_length):
```

or even

```suggestion
    def _repeat_and_group_metapaths(self, metapaths, walk_length):
```

Reply:
`extend` sounds good to me. `augment` came from me taking the phrasing from an existing comment that I removed `# augment metapath to be length long`.